### ISSUE TITLE
Quoting Charsets (Trac #67)

### ISSUE THREAD

**Issue by mnot** [httpwg/httpbis-issues#67]
The BNF definition of the "Content-Type" production does not utiilize the "charset" production defined in section 3.4, and therefore an occasional reader of RFC2616 who follows the BNF dependencies, does not necessarily notice section 3.4 and hence arrives at the conclusion that the quoted and unquoted form are both equally ok.

Migrated from https://trac.ietf.org/ticket/67
```json
{
    "status": "closed",
    "changetime": "2008-07-30T06:45:22",
    "_ts": "1217400322000000",
    "description": "The BNF definition of the \"Content-Type\" production does not utiilize the \"charset\" production defined in section 3.4, and therefore an occasional reader of RFC2616 who follows the BNF dependencies, does not necessarily notice section 3.4 and hence arrives at the conclusion that the quoted and unquoted form are both equally ok.",
    "reporter": "mnot@pobox.com",
    "cc": "",
    "resolution": "fixed",
    "time": "2007-12-22T02:53:13",
    "component": "p3-payload",
    "summary": "Quoting Charsets",
    "priority": "",
    "keywords": "",
    "milestone": "03",
    "owner": "julian.reschke@gmx.de",
    "type": "design",
    "severity": ""
}
```

**Comment by mnot** [comment]
Trac update at 20071222T02:53:41: mnot@pobox.com commented:

> I suggest to fix this by utilizing the "charset" production somewhere in the "Content-Type" production. Maybe at the level of "media-type". In addition, an explicit reference to section 3.4 could be added to the description of Content-Type in section 14.17.

**Comment by mnot** [comment]
Trac update at 20080104T06:05:40: 
* mnot@pobox.com changed component from "" to "messaging"
* mnot@pobox.com changed milestone from "" to "unassigned"
* mnot@pobox.com changed owner from "" to ""

**Comment by mnot** [comment]
Trac update at 20080105T01:53:54: 
* fielding@gbiv.com changed milestone from "unassigned" to "02"
* fielding@gbiv.com changed owner from "" to "fielding@gbiv.com"
* fielding@gbiv.com changed status from "new" to "assigned"

**Comment by mnot** [comment]
Trac update at 20080205T01:38:02: 
* fielding@gbiv.com commented:

> There is some confusion here.  First, HTTP allows both quoted and unquoted
> forms in Content-Type, and that certainly isn't going to change.  However,
> HTTP only uses the charset ABNF production in Accept-Charset, and thus
> is currently defined to only allow tokens in Accept-Charset.
> 
> Should Accept-Charset allow charset quoted strings?  I don't think so.
> Should the charset production be removed to reduce the confusion?
> Perhaps.  This is really a design issue.
> 
> This would be a lot easier if IANA kept a decent registry for charset
> that only included the "MIME preferred names".  We may need to request
> that in the IANA considerations.


* fielding@gbiv.com changed type from "editorial" to "design"

**Comment by mnot** [comment]
Trac update at 20080228T05:26:57: mnot@pobox.com changed milestone from "02" to "03"

**Comment by mnot** [comment]
Trac update at 20080402T06:00:44: mnot@pobox.com changed milestone from "03" to "unassigned"

**Comment by mnot** [comment]
Trac update at 20080509T06:05:02: mnot@pobox.com commented:

> Proposal:
>   - leave accept-charset as-is.
>   - explicitly allow all media-type parameters defined as a token to occur as a quoted-string, and make them equivalent (p2 section 3.3).

**Comment by mnot** [comment]
Trac update at 20080526T11:42:08: 
* mnot@pobox.com commented:

> Proposal accepted.


* mnot@pobox.com changed milestone from "unassigned" to "03"

**Comment by mnot** [comment]
Trac update at 20080528T14:55:30: julian.reschke@gmx.de commented:

> From [250]:
> 
> Improve alignment in BNF (related to #67)

**Comment by mnot** [comment]
Trac update at 20080528T15:46:03: 
* julian.reschke@gmx.de changed component from "p1-messaging" to "p3-payload"
* julian.reschke@gmx.de changed owner from "fielding@gbiv.com" to "julian.reschke@gmx.de"
* julian.reschke@gmx.de changed status from "assigned" to "new"

**Comment by mnot** [comment]
Trac update at 20080529T15:50:47: 
* julian.reschke@gmx.de changed attachment from "" to "i67.diff"
* julian.reschke@gmx.de commented:

> proposed change for part 3

**Comment by mnot** [comment]
Trac update at 20080530T09:52:37: 
* julian.reschke@gmx.de commented:

> Fixed in [251]:
> 
> Resolve #67: media-range: token and quoted-string notation for parameters is equivalent (closes #67).


* julian.reschke@gmx.de changed resolution from "" to "fixed"
* julian.reschke@gmx.de changed status from "new" to "closed"

**Comment by mnot** [comment]
Trac update at 20080730T04:13:02: 
* fielding@gbiv.com changed attachment from "" to "i67-roy.diff"
* fielding@gbiv.com commented:

> Rephrase to focus on parameter values, not parameters.

**Comment by mnot** [comment]
Trac update at 20080730T04:15:43: 
* fielding@gbiv.com commented:

> The change in [251] incorrectly refers to parameters instead of
> parameter values.  I have rephrased it in the second attachment.
> 


* fielding@gbiv.com changed resolution from "fixed" to ""
* fielding@gbiv.com changed status from "closed" to "reopened"

**Comment by mnot** [comment]
Trac update at 20080730T06:45:22: 
* julian.reschke@gmx.de commented:

> Fixed in [297]:
> 
> Resolve #67: Adopt rephrasing of the statement proposed by Roy (closes #67).


* julian.reschke@gmx.de changed resolution from "" to "fixed"
* julian.reschke@gmx.de changed status from "reopened" to "closed"
